scripts/bash_lib.sh

```diff
@@ -528,4 +537,133 @@ check_upgraded_versions() {
     bb_log_err "or if it is a development tree that is based on an old version"
     exit 1
   fi
+
+  res=0
```

Comment:
@elenst 

This block of code compares the differences in engines for both major and minor upgrades, right?

I'm wondering what our policy is in this case—specifically, whether we allow differences in the engine list between major versions.

I'm not sure if we remove engines, but what if we add new ones? This check wouldn’t be future-proof in that case.

Reply:
We do remove engines, but it happens extremely rarely (more rarely than it should, in my opinion). Last time it happened to my remembrance was when we disabled TokuDB in 10.5. We do, however, lose them from time to time inadvertently -- also extremely rarely, but it happened. So, I think it should be a fairly low-maintenance check for major upgrades -- even if we do remove an engine once in several years, or an engine's support flag changes (XA, transactions), an exception will be easy to add. 

Adding engines can happen, that's why the check is only for "disappeared or changed" (comm -23).

Reply:
Sure, better to be on the safe side and do the checks for both.